x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/multiterms/MultiTermsAggregator.java

```diff
@@ -95,11 +95,14 @@ protected MultiTermsAggregator(
         partiallyBuiltBucketComparator = order == null ? null : order.partiallyBuiltBucketComparator(b -> b.bucketOrd, this);
         this.formats = formats;
         this.showTermDocCountError = showTermDocCountError;
-        if (subAggsNeedScore() && descendsFromNestedAggregator(parent)) {
+        if (subAggsNeedScore() && descendsFromNestedAggregator(parent) || context.isInSortOrderExecutionRequired()) {
```

Comment:
So if I understand correctly here we are forcing depth first mode when we have a time series aggregation so to avoid using breadth first and computing an incorrect result (causes by the usage of deferred collector in case of depth first).

Reply:
Should add a test where we check that execution of multi terms fails if the user specifies `collect_mode: breadth_first` and uses a time series aggregation together with a multi terms aggregation?

Reply:
Yes, that is correct we enforce depth first. This is consistent with the behaviour in terms aggregation.